### PR TITLE
TagBox - mark selected items in filtered results (T880346)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 1]
```diff
@@ -3188,6 +3190,33 @@ QUnit.module('searchEnabled', moduleSetup, () => {
...
+            showSelectionControls: true,
+            deferRendering: true,
+            minSearchLength: 2
+        });
+        const instance = $element.dxTagBox('instance');
+        const $input = $element.find(`.${TEXTBOX_CLASS}`);
+
+        keyboardMock($input)
+            .type('aa');
+        $(instance.content()).find(`.${LIST_CKECKBOX_CLASS}`).eq(0).trigger('dxclick');
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]:
```suggestion
        $(`.${LIST_CKECKBOX_CLASS}`).eq(0).trigger('dxclick');
```
Why not like [here](https://github.com/DevExpress/DevExtreme/pull/12874/files#diff-2da7b06a8e2cf07b9cb96e242b9fba75R3214)?

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 2]
```diff
@@ -3188,6 +3190,33 @@ QUnit.module('searchEnabled', moduleSetup, () => {
...
+        $(instance.content()).parent().find(`.${POPUP_DONE_BUTTON_CLASS}`).trigger('dxclick');
+
+        keyboardMock($input)
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: Let's reuse keyboardMock from [here](
https://github.com/DevExpress/DevExtreme/pull/12874/files#diff-2da7b06a8e2cf07b9cb96e242b9fba75R3207).

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 3]
```diff
@@ -3188,6 +3190,33 @@ QUnit.module('searchEnabled', moduleSetup, () => {
...
+            deferRendering: true,
+            minSearchLength: 2
+        });
+        const instance = $element.dxTagBox('instance');
+        const $input = $element.find(`.${TEXTBOX_CLASS}`);
+
+        keyboardMock($input)
+            .type('aa');
+        $(instance.content()).find(`.${LIST_CKECKBOX_CLASS}`).eq(0).trigger('dxclick');
+        $(instance.content()).parent().find(`.${POPUP_DONE_BUTTON_CLASS}`).trigger('dxclick');
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]:
```suggestion
        $(`.${POPUP_DONE_BUTTON_CLASS}`).trigger('dxclick');
```
?

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 4]
```diff
@@ -3188,6 +3190,33 @@ QUnit.module('searchEnabled', moduleSetup, () => {
...
+
+        keyboardMock($input)
+            .type('aa');
+        $(`.${LIST_CKECKBOX_CLASS}`).eq(1).trigger('dxclick');
+        $(instance.content()).parent().find(`.${POPUP_DONE_BUTTON_CLASS}`).trigger('dxclick');
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]:
```suggestion
        $(`.${POPUP_DONE_BUTTON_CLASS}`).trigger('dxclick');
```
?

**Review by DokaRus — DISMISSED** [review]

**Review by San4es — APPROVED** [review]
LGTM

**Review by DokaRus — APPROVED** [review]

**Pull request merged**